test/sql/aggregate/aggregates/test_string_agg.test

```diff
@@ -88,3 +91,58 @@ SELECT g, GROUP_CONCAT(x) FROM strings GROUP BY g ORDER BY g
 2	i,j
 3	p
 4	x,y,z
+
+#
+# Test ORDER BY
+#
+
+# Impervious to threading
+statement ok
+PRAGMA threads=4
+
+# Single group
+query TT
+SELECT STRING_AGG(x,',' ORDER BY x ASC), STRING_AGG(x,y ORDER BY x ASC) FROM strings
+----
+a,b,i,j,p,x,y,z
+a-b/i+j/p/x-y+z
+
+query TT
+SELECT STRING_AGG(x,',' ORDER BY x DESC), STRING_AGG(x,y ORDER BY x DESC) FROM strings
+----
+z,y,x,p,j,i,b,a
+z-y/x/p+j/i-b/a
+
+# Constant separator
+query II
+SELECT g, STRING_AGG(x, ',' ORDER BY x ASC) FROM strings GROUP BY g ORDER BY 1
+----
+1	a,b
+2	i,j
+3	p
+4	x,y,z
+
+query II
+SELECT g, STRING_AGG(x, ',' ORDER BY x DESC) FROM strings GROUP BY g ORDER BY 1
```

Comment:
Can we add a test that runs ORDER BY on a correlated column, e.g.:

```sql
CREATE TABLE integers(i INTEGER);
INSERT INTO integers VALUES (1), (2), (3), (NULL);
SELECT i1.i, (SELECT STRING_AGG(i::VARCHAR, ',' ORDER BY i1.i) FROM integers WHERE i<=i1.i) c1 FROM integers i1;

-- result from postgres
 i |  c1   
---+-------
 1 | 1
 2 | 1,2
 3 | 1,2,3
NULL | NULL

```

Reply:
Well that crashes! No idea why - it all gets translated into expression arguments.

Reply:
I suspect that is because the new order expressions are not added to the `duckdb/parser/parsed_expression_iterator.hpp` and `duckdb/planner/expression_iterator.hpp` classes, which results in the subquery expressions not being flattened properly.

Reply:
I don't think `ExpressionIterator` needs to be changed because the expressions have already been moved to the `children`.

Reply:
`SelectBinder::BindAggregate` needs to `BindCorrelatedColumns` on the `ORDER BY` expressions too.